src/cascadia/TerminalControl/XamlLights.cpp

```diff
@@ -49,59 +46,66 @@ namespace winrt::Microsoft::Terminal::Control::implementation
+            _IsTargetProperty =
+                DependencyProperty::RegisterAttached(
+                    L"IsTarget",
+                    winrt::xaml_typename<bool>(),
+                    winrt::xaml_typename<Control::CursorLight>(),
+                    PropertyMetadata{ winrt::box_value(false), PropertyChangedCallback{ &CursorLight::OnIsTargetChanged } });
         }
     }
 
-    winrt::hstring VisualBellLight::GetId()
+    void CursorLight::ChangeLocation(float xCoord, float yCoord)
     {
-        return VisualBellLight::GetIdStatic();
+        if (const auto light = CompositionLight())
+        {
+            if (const auto cursorLight = light.try_as<Windows::UI::Composition::SpotLight>())
+            {
+                cursorLight.Offset({ xCoord, yCoord, 100 });
+            }
+        }
+        else
+        {
+            _InitializeHelper(xCoord, yCoord);
+        }
     }
 
-    void VisualBellLight::OnIsTargetChanged(DependencyObject const& d, DependencyPropertyChangedEventArgs const& e)
+    // Method Description:
+    // - This function is called when the first target UIElement is shown on the screen,
+    //   this enables delaying composition object creation until it's actually necessary.
+    // Arguments:
+    // - newElement: unused
+    void CursorLight::OnConnected(UIElement const& /* newElement */)
     {
-        const auto uielem{ d.try_as<UIElement>() };
-        const auto brush{ d.try_as<Brush>() };
-
-        if (!uielem && !brush)
+        if (!CompositionLight())
         {
-            // terminate early
-            return;
+            _InitializeHelper(0, 0);
         }
+    }
 
-        const auto isAdding = winrt::unbox_value<bool>(e.NewValue());
-        const auto id = GetIdStatic();
-
-        if (isAdding)
-        {
-            if (uielem)
-            {
-                XamlLight::AddTargetElement(id, uielem);
-            }
-            else
-            {
-                XamlLight::AddTargetBrush(id, brush);
-            }
-        }
-        else
+    // Method Description:
+    // - Helper to initialize the properties of the spotlight such as the location and
+    //   the angles of the inner and outer cones
+    // Arguments:
+    // - xCoord: the x-coordinate of where to put the light
+    // - yCoord: the y-coordinate of where to put the light
+    void CursorLight::_InitializeHelper(float xCoord, float yCoord)
+    {
+        if (!CompositionLight())
         {
-            if (uielem)
-            {
-                XamlLight::RemoveTargetElement(id, uielem);
-            }
-            else
-            {
-                XamlLight::RemoveTargetBrush(id, brush);
-            }
+            auto spotLight{ Window::Current().Compositor().CreateSpotLight() };
+            spotLight.InnerConeColor(Windows::UI::Colors::White());
+            spotLight.InnerConeAngleInDegrees(10);
+            spotLight.OuterConeAngleInDegrees(25);
+            spotLight.Offset({ xCoord, yCoord, 100 });

```

Comment:
So if we did want to have the outer area not fully opaque black, then we could set `OuterConeColor` here. Neat!

Reply:
We'd actually need to change the `OuterConeAngle` to be wider, I gave it a shot after the suggestion during the share-out but I really didn't like it so I'm keeping it the way it is. Here's what it looked like:
<img width="806" alt="light" src="https://user-images.githubusercontent.com/26824113/128779623-c9a1e4ad-a1c1-4581-ab88-b42d78fa1dd8.png">